src/_pytest/config/findpaths.py

```diff
@@ -161,24 +161,26 @@ def determine_setup(
     args: List[str],
     rootdir_cmd_arg: Optional[str] = None,
     config: Optional["Config"] = None,
-) -> Tuple[py.path.local, Optional[str], Dict[str, Union[str, List[str]]]]:
+) -> Tuple[py.path.local, Optional[py.path.local], Dict[str, Union[str, List[str]]]]:
     rootdir = None
     dirs = get_dirs_from_args(args)
     if inifile:
-        inicfg = load_config_dict_from_file(py.path.local(inifile)) or {}
+        inipath_ = py.path.local(inifile)
```

Comment:
I'm curious, why this `inipath_/inipath` dance here? 🤔

Reply:
If I had used `inipath = ... # type: Optional[py.path.local]` here, then I wouldn't be able to pass it to `load_config_dict_from_file` which requires `py.path.local`.

In Python 3.6 I would have declared `inipath: Optional[py.path.local]` at the top of the function but for now we need to use the comments.

Reply:
Not sure I follow, don't `inipath` and `inipath_` point to the same object? I mean, `inipath_` is a `py.path.local` object, how can `inipath = inipath_` be typed as `Optional[py.path.local]`?

Reply:
Stripping to only the relevant bits:

```py
if inifile:
    inipath = py.path.local(inifile)  # A
    inicfg = load_config_dict_from_file(inipath)  # B
else:
   inipath = something that returns Optional[py.path.local]  # C
```

As written, A will set the type of `inipath` as `py.path.local`, and then C is invalid.

The usual solution to this is to change A to

```py
    inipath = py.path.local(inifile) # type: Optional[py.path.local]
```

but this causes a new problem in line B, that the argument is required to be `py.path.local` but is now `Optional[py.path.local]`.

So I worked around it with the `inipath_`. Let me know if you think of a better way :)

Reply:
Thanks for taking the time to explain, appreciate it!